src/vs/workbench/browser/parts/editor/editorParts.ts

```diff
@@ -387,6 +389,12 @@ export class EditorParts extends MultiWindowParts<EditorPart> implements IEditor
 			return false;
 		}
 
+		// Applying a working set can be the result of a user action that has been
+		// initiated from the terminal (ex: switching branches). As such, we want
+		// to preserve the focus in the terminal. This does not cover the scenario
+		// in which the terminal is in the editor part.
+		const preserveFocus = this.layoutService.hasFocus(Parts.PANEL_PART);
```

Comment:
I was expecting a `options?: { preserveFocus?: boolean }` in `applyWorkingSet` so that the caller controls this.